src/MCPClient/lib/clientScripts/hasPackages.py

```diff
@@ -19,13 +24,25 @@ def is_extractable(f):
     else:
         return False
 
+def already_extracted(f):
+    """
+    Returns True if this package has already been extracted, False otherwise.
+    """
+    # Look for files in a directory that starts with the package name
+    files = File.objects.filter(transfer=f.transfer, currentlocation__startswith=f.currentlocation, removedtime__isnull=True).exclude(uuid=f.uuid)
+    # Check for unpacking events that reference the package
+    if Event.objects.filter(file_uuid__in=files, event_type='unpacking', event_detail__contains=f.currentlocation).exists():
```

Comment:
If we don't already, let's start tracking parentage information and use that here - e.g. have a way to filter on "unpacked from X", which we could use to determine if anything had been extracted from it.